src/handler/batchPlateCreation.ts

```diff
@@ -1,54 +1,48 @@
 import { TechRecordType } from '@dvsa/cvs-type-definitions/types/v3/tech-record/tech-record-verb';
 import 'dotenv/config';
 import { v4 as uuidv4 } from 'uuid';
-import { PlateReasonForIssue, Plates } from '../../models/plate';
-import { DocumentName, SQSRequestBody } from '../../models/sqsPayload';
-import { getBySystemNumberAndCreatedTimestamp, inPlaceRecordUpdate } from '../../services/database';
-import { addToSqs } from '../../services/sqs';
-import { ERRORS, StatusCode } from '../../util/enum';
-import { formatErrorMessage } from '../../util/errorMessage';
-import { flattenArrays, formatTechRecord } from '../../util/formatTechRecord';
-import { addHttpHeaders } from '../../util/httpHeaders';
-import logger from '../../util/logger';
-import batchPlatesData from './resources/batchPlatesData.json';
-
-export type BatchPlateData = {
-  systemNumber: string,
-  createdTimestamp: string
-};
-
-export const handler = async (batchPlateRecords: BatchPlateData[]) => {
+import { SQSEvent } from 'aws-lambda';
+import { PlateReasonForIssue, Plates } from '../models/plate';
+import { DocumentName, SQSRequestBody } from '../models/sqsPayload';
+import { getBySystemNumberAndCreatedTimestamp, inPlaceRecordUpdate } from '../services/database';
+import { addToSqs } from '../services/sqs';
+import { ERRORS, StatusCode } from '../util/enum';
+import { formatErrorMessage } from '../util/errorMessage';
+import { flattenArrays, formatTechRecord } from '../util/formatTechRecord';
+import logger from '../util/logger';
+import { BatchPlateData } from '../models/batchPlate';
+
+export const handler = async (event: SQSEvent): Promise<void> => {
   try {
     logger.info('Batch plate generation called');
 
-    const batchData = batchPlateRecords ?? batchPlatesData;
-
     const batchIssuerName = 'CVS Batch Plate Generation';
     let numberOfRecordsUpdated = 0;
     let numberOfSqsAdded = 0;
 
-    // eslint-disable-next-line no-restricted-syntax
-    for await (const data of batchData) {
+    const processPromises = event.Records.map(async (sqsRecord) => {
+      const { body } = sqsRecord;
+      const data: BatchPlateData = JSON.parse(body) as BatchPlateData;
       const { systemNumber, createdTimestamp } = data;
+
       logger.info(`Get from database with sysNum ${systemNumber} and timestamp ${createdTimestamp}`);
 
       try {
-        const record = await getBySystemNumberAndCreatedTimestamp(systemNumber, createdTimestamp);
-        logger.debug(`result is: ${JSON.stringify(record)}`);
+        const dbRecord = await getBySystemNumberAndCreatedTimestamp(systemNumber, createdTimestamp);
+        logger.debug(`result is: ${JSON.stringify(dbRecord)}`);
 
-        if (!record || !Object.keys(record).length) {
+        if (!dbRecord || !Object.keys(dbRecord).length) {
           logger.error(`Missing record with sysNum ${systemNumber} and timestamp ${createdTimestamp}`);
-          continue;
+          return;
```

Comment:
we probably want to consider which of of these should log and return, and which need to actually throw errors and get sent to the DL queue.

Reply:
would we want all of these if statements to throw?

Reply:
Maybe its one for the error handling ticket - But I'd probably say yes. The new DLQ is set up to not retry, I believe, so we'd effectively be throwing these to somewhere that we could see which records were failing